### ISSUE TITLE
glue within data.table

### ISSUE THREAD

**Issue by jakejh** [hugheylab/actions#2]
Is this really the most concise way to do it? It hurts to look at it. ;-)

https://github.com/hugheylab/actions/blob/90c5242e776f05bf75000ab509501ed5af43f0a8/workflows/custom-lint.R#L79

**Comment by JSchoenbachler** [comment]
I tried using the `.envir = .SD` and that isn't working when I call it as a function for some reason. This was the easiest way to get around that, but I can look into it some more and see if I can figure it out.

This is the error that I get when I try to run using the .envir variable:

```
Error in new.env(parent = eval_env) : 'enclos' must be an environment 
```

**Comment by jakejh** [comment]
Hmm, that's curious. I'd rather use `sprintf` than the current implementation. When you tried .envir, was every single variable in .SD?

**Comment by JSchoenbachler** [comment]
So an update to this:

The error I first posted 

```
Error in new.env(parent = eval_env) : 'enclos' must be an environment 
``` 

only happened when I tried it like this:

```
lfDt[, format_line :=
  glue('{.I}. {filename} line {line_number}: {message} ({lint_link}){newLineEsc}    ```r{newLineEsc}    {line}{newLineEsc}    ```',
  newLineEsc = ' \r\n', .envir = .SD)]
```

Whereas when I removed the `newLineEsc = ' \r\n`` and did this:

```
lfDt[, format_line :=
  glue('{.I}. {filename} line {line_number}: {message} ({lint_link}) \r\n   ```r \r\n   {line} \r\n   ```',
  .envir = .SD)]
```

the column just came back as an NA string.

I can just use `sprintf`.

**Comment by JSchoenbachler** [comment]
Resolved with 459c601a348f39750829d3c5c59e67378479f08f
